### ISSUE TITLE
Add more buttons

### ISSUE THREAD

**Issue by felipenmoura** [NascHQ/nasc-vscode-mac-touchbar#2]
Once we got support for the user to customize the buttons (still depending on some fixes in the touchbar support for the vsCode), we could have some extra buttons:

- [ ] Rename file (already has an open PR)
- [ ] Duplicate selection/line
- [ ] Find next
- [ ] Toggle "show white spaces"
- [ ] Fold/unfold

Any other suggestion?

**Comment by daveshirman** [comment]
- find all references
- peek definition
- close all windows
- close others
- snap window left (internally to app - not OS)
- snap window right (internally to app - not OS)

**Comment by wheelhot** [comment]
Trigger intellisense manually, useful for knowing CSS property has what values.

Right now the shortcut key is control+spacebar

**Comment by felipenmoura** [comment]
Just published an update adding quite a few extra tools.
I hope you enjoy it.

Again, open for suggestions for features (or perhaps, better symbols to use as icons)

**Comment by wheelhot** [comment]
Odd, I don't see the update in VSCode

**Comment by felipenmoura** [comment]
hm, odd it is!
Even if you go to the list of installed extensions? There should be a button for you to look for updates there.
When you look for the extension, do you see the updated readme file with a few extra screenshots?

**Comment by daveshirman** [comment]
Is it easy enough to add these two options?

1. snap tab left (internally to app - not OS)
2. snap tab right (internally to app - not OS)

**Comment by felipenmoura** [comment]
Sure, why not.
But I'm not sure what are those!
I thought those where "increase/decrease indentation".
In the list here:
https://code.visualstudio.com/docs/getstarted/keybindings#_when-clause-contexts

Which one is what you are looking for?

**Comment by daveshirman** [comment]
I don't know if there is a shortcut for it - it's when you drag a tab to
the left or right and it fills half the window, like windows does with
window snapping, but inside the app itself.

**Comment by felipenmoura** [comment]
hm, interesting.
I think that is the "split editor", isn't it?
I'm guessing these would be the options you are looking for!
You want to make it happen and open a PR?
If not, I might have time to implement it next weekend.

<img width="750" alt="screen shot 2017-11-13 at 20 20 39" src="https://user-images.githubusercontent.com/347387/32752405-31df1e14-c8b0-11e7-8247-79a94bca1bf9.png">

**Comment by daveshirman** [comment]
Yep those look like the ones.  Again, I have no time outside of my projects
sorry, all yours! Loving your work :)

**Comment by OliverUrbann** [comment]
I would like to create buttons that just insert some text, e.g. "def". On a german keyboard I have to press option-5 for a [ which is not convenient, so I would also like to have a button for brackets etc.

**Comment by felipenmoura** [comment]
hm, interesting...we could think of a button that could add whatever you want, so you could configure it in the settings...would that help?

**Comment by OliverUrbann** [comment]
Actually some hours later I've found BetterTouchTool which exactly provides this feature, sending arbitrary keystrokes with configurable button images. However, it is not context sensitive, so e.g. in my PyCharm setup I have always the debug buttons (step etc.) on the Touch Bar. Here my current setup:

<img width="1085" alt="touchbar" src="https://user-images.githubusercontent.com/13621807/33514668-40631a92-d758-11e7-9c7a-91823fd00006.png">

**Comment by felipenmoura** [comment]
Interesting @OliverUrbann and what plugin is that? We might get some inspiration from it (specially if it is open source).

**Comment by OliverUrbann** [comment]
It's not open source and it is not a plugin. It is a general tool for customizing gestures, touch bar etc. in every app.

**Comment by felipenmoura** [comment]
I see.
Ok, let's see if we can add this feature here, so developers will have their custom controls contextualized to VSCode.

**Comment by liqwid** [comment]
VSCode reload button( linters/git/jest freeze sometimes) would be nice to have

**Comment by benny-gold** [comment]
`Run Selection` (F8) would be really useful

**Comment by eisenheim94** [comment]
Would be cool to add support for Git push/pull button

![Git push/pull](https://i.imgur.com/ldJSGzx.png)

**Comment by felipenmoura** [comment]
Interesting, you mean "run selected text in terminal"?

**Comment by benny-gold** [comment]
Yes and no. Selfishly, I want `Powershell.RunSelection` but the command for other languages is `workbench.action.terminal.runSelectedText` I created a [fork ](https://github.com/benny-gold/nasc-vscode-mac-touchbar) to try and add it myself but this is my first extension so it's a bit of trial & error at the moment. Watch this space!

**Comment by piyushgupta1** [comment]
Organise imports and code auto formatting would also be useful

**Comment by felipenmoura** [comment]
> Yes and no. Selfishly, I want `Powershell.RunSelection` but the command for other languages is `workbench.action.terminal.runSelectedText` I created a [fork ](https://github.com/benny-gold/nasc-vscode-mac-touchbar) to try and add it myself but this is my first extension so it's a bit of trial & error at the moment. Watch this space!

Did you get it working, @benny-gold ?

**Comment by robjtede** [comment]
+1 on `find all references` option, the key combo Option+Shift(+Fn)+F12 is kinda awkward.
